recursion.rb

```diff
@@ -33,8 +78,55 @@ def nested(s)
 puts "passes all bunny tests"
 
 # Nested Tests
-raise "nested broke - nested('((()))')" unless reverse("((()))") == true
-raise "nested broke - nested('())')" unless reverse("())") == false
+raise "nested broke - nested('((()))')" unless nested("((()))") == true
+raise "nested broke - nested('())')" unless nested("())") == false
 puts "passes all nested tests"
 
+
+
+
+
+### Added Fun ###
+
+# ### fib(n)
+# Write a recursive method `fib` that accepts an integer n as a parameter and returns the nth [fibonacci number](https://en.wikipedia.org/wiki/Fibonacci#Fibonacci_sequence).
+#
+# - e.g. fib(4) = (1 1 2) 3 = 3
+#1 1 2 3 5 8 13 21 34 55
+
+def fib(n)
+  if n <= 2
+    return 1
+  else
+    return fib(n-2) + fib(n-1)
+  end
+end
+
+raise "fib broke - fib(5)" unless fib(5) == 5
+raise "fib broke - fib(8)" unless fib(8) == 21
+puts "passes all fib tests"
+
+
+
+# ### pal(s)
+# Write a recursive method `pal` that accepts a string s as a parameter and returns a boolean value indicating if that string is a [palindrome](https://en.wikipedia.org/wiki/Palindrome) or not.
+#
+# - e.g. pal("racecar") = true
+# - e.g. pal("smile") = false
+
+def pal(string)
+  if string == ""
+    return true
+  elsif string[0] == string[-1]
+    return pal(string[1..-2])
+  else
+    return false
+  end
+end
```

Comment:
This looks good - I think the only thing that I'd suggest is not using `string` as a variable name. In strongly-typed languages, you have to specify the data type when declaring a variable (e.g. in java `String str` where `str` is the name of the variable)  so naming a variable with its data type is confusing. For a string, I'd use `str` or `s` instead.